core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfiguration.java

```diff
@@ -66,35 +47,34 @@ public AsyncStoreConfiguration async() {
     */
    @Override
    public boolean purgeOnStartup() {
-      return purgeOnStartup.get();
+      return attributes.attribute(PURGE_ON_STARTUP).get();
    }
 
    @Override
    public boolean shared() {
-      return shared.get();
+      return attributes.attribute(SHARED).get();
    }
 
    @Override
    public boolean transactional() {
-      return transactional.get();
+      return attributes.attribute(TRANSACTIONAL).get();
    }
 
    @Override
    public int maxBatchSize() {
-      return maxBatchSize.get();
+      return attributes.attribute(MAX_BATCH_SIZE).get();
```

Comment:
Kind of a stretch and does not need to apply. Now we're doing a map lookup, and this method is invoked within a loop on `RemoteStore:388`, so we could change there to hold the value on a variable instead.

Reply:
Since we may want to allow mutation of this, I'm thinking of leaving it as-is here and extracting it from the loop in `RemoteStore` and `ComposedSegmentedLoadStore`

Reply:
Aside from the above change, I just noticed that `SchemaTableOperations` holds on to `maxBatchSize`, `writeQueryTimeout` and `readQueryTimeout` which means it won't be aware of any changes to these. I've addressed this too.